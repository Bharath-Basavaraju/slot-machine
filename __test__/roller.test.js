import {Map} from "immutable"
import RollerComponent from "../src/view/roller";
import React from 'react';
import {shallow} from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

function setup(props){
    const wrapper = shallow(<RollerComponent {...props} />);
    return {
        props,
        wrapper
    }
}

describe("tests for Roller (wheel) component", () => {
    it("Should load properly for roller(wheel) data when there's no backgroundPosition", () => {
        const elementParams = {
            "elementData": {
                "name": 'orange',
                "id": "2"
            }
        };
        const { wrapper } = setup(elementParams);
        expect(shallowToJson(wrapper)).toMatchSnapshot("test for backgroundPosition is missing");
    });
    it("Should load properly for roller(wheel) element data", () => {
        const elementParams = {
            "elementData": {
                "name": 'orange',
                "id": "2",
                "backgroundPosition": "-0 -200px"
            }
        };
        const { wrapper } = setup(elementParams);
        expect(shallowToJson(wrapper)).toMatchSnapshot("test for backgroundPosition is present in element data");
    });
});