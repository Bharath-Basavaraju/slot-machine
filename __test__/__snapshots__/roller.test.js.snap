// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Roller (wheel) component Should load properly for roller(wheel) data when there's no backgroundPosition: test for backgroundPosition is missing 1`] = `
<div
  style={
    Object {
      "backgroundPosition": "0 0",
    }
  }
/>
`;

exports[`tests for Roller (wheel) component Should load properly for roller(wheel) element data: test for backgroundPosition is present in element data 1`] = `
<div
  style={
    Object {
      "backgroundPosition": "-0 -200px",
    }
  }
/>
`;
